config/worker.js

```diff
@@ -0,0 +1,146 @@
+const CLIENT_ID = '6caaf5d0a58e47868553160955e5b8e4'
+const CLIENT_SECRET = 'REDACTED'
+const REDIRECT_URL = '/spotifycb'
+const WORKER_URL = 'https://duoco.cawman.workers.dev'
+const CALLBACK_URL = 'file://callback'
+
+const refreshtokensMap = new Map()
+
+/**
+ * Example of how router can be used in an application
+ *  */
+addEventListener('fetch', event => {
+    event.respondWith(handleRequest(event.request))
+})
+
+
+async function handleRequest(request) {
+    let requestURL = new URL(request.url)
+    
+    switch(requestURL.pathname) {  
+        case '/spotifylogin': {
+            let scopes = 'user-read-currently-playing'
+            //let referrer = requestURL.searchParams.get('ref')
+            let url = 'https://accounts.spotify.com/authorize' +
```

Comment:
Told you already, you shouldn't be hardcoding the API url. It should be in the config file.